File: refactored/sleepy-test/CSharpXUnit/CSharpXUnit.Tests/Tests/CleanClassNamesOfAnonymousDelegatesTest.cs
using Xunit;

namespace DefaultNamespace
{
    public class CallSiteTests
    {
        [Fact]
        public void CleanClassNamesOfAnonymousDelegatesTest()
        {
            {
                var logFactory = new LogFactory().Setup().LoadConfigurationFromXml(@"
                            <nlog>
                                <targets><target name='debug' type='Debug' layout='${callsite:ClassName=true:MethodName=false:CleanNamesOfAnonymousDelegates=true}' /></targets>
                                <rules>
                                    <logger name='*' levels='Fatal' writeTo='debug' />
                                </rules>
                            </nlog>").LogFactory;
                var logger = logFactory.GetLogger("A");
                bool done = false;
                ThreadPool.QueueUserWorkItem(state =>
                {
                    logger.Fatal("message");
                    done = true;
                }, null);
                while (done == false)
                {
                    // Thread.Sleep(10);
                }
            
                if (done == true)
                {
                    logFactory.AssertDebugLastMessage("NLog.UnitTests.LayoutRenderers.CallSiteTests");
                }
            }
        }
    }
}

File: sleepy-test/java-junit/CleanClassNamesOfAnonymousDelegatesTest.java
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;
import java.util.concurrent.atomic.AtomicBoolean;

public class CallSiteTests {

    @Test
    public void cleanClassNamesOfAnonymousDelegatesTest() {
        var logFactory = new LogFactory().setup().loadConfigurationFromXml(
                "<nlog>" +
                "<targets><target name='debug' type='Debug' layout='${callsite:ClassName=true:MethodName=false:CleanNamesOfAnonymousDelegates=true}' /></targets>" +
                "<rules>" +
                "<logger name='*' levels='Fatal' writeTo='debug' />" +
                "</rules>" +
                "</nlog>").logFactory();
        var logger = logFactory.getLogger("A");
        AtomicBoolean done = new AtomicBoolean(false);
        Thread thread = new Thread(() -> {
            logger.fatal("message");
            done.set(true);
        });
        thread.start();
        
        while (!done.get()) {
            try {
                Thread.sleep(10);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }

        if (done.get()) {
            logFactory.assertDebugLastMessage("NLog.UnitTests.LayoutRenderers.CallSiteTests");
        }
    }
}
